pkg/syncer/cnsoperator/controller/cnsfileaccessconfig/cnsfileaccessconfig_controller.go

```diff
@@ -0,0 +1,353 @@
+	vmOperatorClient client.Client
+	recorder         record.EventRecorder
+}
+
+// Reconcile reads that state of the cluster for a CnsFileAccessConfig object and makes changes based on the state read
+// and what is in the CnsFileAccessConfig.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileCnsFileAccessConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	ctx, log := logger.GetNewContextWithLogger()
+	// Fetch the CnsFileAccessConfig instance
+	instance := &cnsfileaccessconfigv1alpha1.CnsFileAccessConfig{}
+	err := r.client.Get(ctx, request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Infof("CnsFileAccessConfig resource not found. Ignoring since object must be deleted.")
+			return reconcile.Result{}, nil
+		}
+		log.Errorf("Error reading the CnsFileAccessConfig with name: %q on namespace: %q. Err: %+v",
+			request.Name, request.Namespace, err)
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Initialize backOffDuration for the instance, if required.
+	backOffDurationMapMutex.Lock()
+	var timeout time.Duration
+	if _, exists := backOffDuration[instance.Name]; !exists {
+		backOffDuration[instance.Name] = time.Second
+	}
+	timeout = backOffDuration[instance.Name]
+	backOffDurationMapMutex.Unlock()
+
+	if instance.DeletionTimestamp != nil {
+		// TODO:
+		// Identify the network provider (NSX-T or VDS).
+		// Find external facing IP of the VM depending on NSXT or VDS - Analysis on how to expose external facing IP of TKG and Pod VMs.
+		// Get ExternalIPtoClientVms[IP] list from FileVolumeClients instance for SV-PVC.
+		// If ExternalIPtoClientVms[IP] exists and CnsFileAccessConfig.Spec.VirtualMachine.Name is the only value in the ExternalIPtoClientVms[IP], then
+		//   Call CNS API to remove the IPAddresses on file volume.
+		// If not,
+		//   Do nothing (as there are other virtual machines on same external IP using this volume)
+		// Call RemoveClientVmFromIPList() to remove req.NodeID to list of nodes exposed by IP referencing this file volume.
+		removeFinalizerFromCRDInstance(ctx, instance)
+		return reconcile.Result{}, nil
+	}
+
+	// If the CnsFileAccessConfig instance is already successful,
+	// and not deleted by the user, remove the instance from the queue.
+	if instance.Status.Done && instance.DeletionTimestamp == nil {
+		// Cleanup instance entry from backOffDuration map
+		backOffDurationMapMutex.Lock()
+		delete(backOffDuration, instance.Name)
+		backOffDurationMapMutex.Unlock()
+		return reconcile.Result{}, nil
+	}
+
+	log.Infof("Reconciling CnsFileAccessConfig with instance: %q from namespace: %q. timeout %q seconds", instance.Name, instance.Namespace, timeout)
+	if !instance.Status.Done && instance.DeletionTimestamp == nil {
```

Comment:
`instance.DeletionTimestamp == nil` can be removed?

Reply:
Makes sense, since we moved the deletion timestamp check to top of the file